packages/xo-web/src/common/xo/index.js

```diff
@@ -1104,7 +1104,9 @@ export const revertSnapshot = vm =>
   )
 
 export const editVm = (vm, props) =>
-  _call('vm.set', { ...props, id: resolveId(vm) })
+  _call('vm.set', { ...props, id: resolveId(vm) }).catch(err => {
+    error(_('setVmFailed', { vm: renderXoItem(vm) }), err.message)
```

Comment:
Are you sure `vm` is always passed as an object? I think it would be safer to use `renderXoItemFromId(resolveId(vm))` here.